### PR TITLE
Add groth16 docs

### PR DESCRIPTION
**Pull request by jonas-lj**

### CODE REVIEW

**Comment by randall-Mysten** [comment]
To publish the topic you must also add it to https://github.com/MystenLabs/sui/blob/main/doc/src/navconfig.json#L109

File: doc/src/learn/cryptography/groth16.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+---
+
+## Overview
+
+A zero-knowledge proof allows a prover to prove that a statement is true without revealing any information about the inputs, for example a prover can prove that they know the solution to a sudoku puzzle without revealing the solution.
+
+Zk-SNARKs (Zero-Knowledge Succinct Non-Interactive Argument of Knowledge) are a family of zero-knowledge proofs which are non-interactive, have succinct proof size and efficient verification time. 
+An important and widely used variant of them is pairing-based zk-SNARKs, of which the [Groth16](https://eprint.iacr.org/2016/260.pdf) proof system is one of the most efficient and widely used.
+
+The Move API in Sui allows users to verify any statement that can be expressed in a NP-complete language can be verified efficiently using Groth16 zk-SNARKs over either the Bn254 or BLS12-381 elliptic curve constructions.
```
Comment (kchalkias) [inline doc/src/learn/cryptography/groth16.md:12]: BN (uppercase as in BLS) - it's initials from author names again.

File: doc/src/learn/cryptography/groth16.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+There are high-level languages for expressing these statements, such as [Circom](https://docs.circom.io) which will be used in the example below.
+
+Note that Groth16 requires a trusted setup for each circuit to generate the verification key, but we will not discuss how to do that here.
```
Comment (kchalkias) [inline doc/src/learn/cryptography/groth16.md:16]: We should write a statement that our api is not pinning any particular VK, each user can generate their own ZK params or use an existing VK to their dapps.

File: doc/src/learn/cryptography/groth16.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+## Usage
+
+The following example demonstrates how to create a Groth16 proof from a statement written in Circom and then verify it using the Sui Move API.
```
Comment (kchalkias) [inline doc/src/learn/cryptography/groth16.md:20]: Add a clear note that our verifier currently supports up to 8 public inputs in the zk circuit. This is important info.

**Review by kchalkias — COMMENTED** [review]
Similarly to the ECVRF comment, we'll need a simple Move smart contract that verifies a ZK in any of the curves (with a comment on how you would it with the other curve too).

**Review by kchalkias — APPROVED** [review]
Awesome! please resolve conflicts and we're good to go!

**Comment by kchalkias** [comment]
> > 
> 
> I resolved them

@randall-Mysten we generally prefer rebasing to merging (to maintain history).
Now I see "This branch cannot be rebased due to conflicts"

**Comment by randall-Mysten** [comment]
> > > 
> > 
> > 
> > I resolved them
> 
> @randall-Mysten we generally prefer rebasing to merging (to maintain history). Now I see "This branch cannot be rebased due to conflicts"

OK

**Pull request merged**